Library/Homebrew/bundle/commands/exec.rb

```diff
@@ -90,7 +101,12 @@ def self.run(*args, global: false, file: nil, subcommand: "", services: false, c
                 rejected_opts = []
                 path = PATH.new(ENV.fetch("PATH"))
                            .reject do |path_value|
-                  rejected_opts << path_value if path_value.match?(opt)
+                  if path_value.match?(opt)
+                    rejected_opts << path_value
+                    true
+                  else
+                    false
+                  end
```

Comment:
Oof,  this block should not be required to explicitly return a boolean, fixed in https://github.com/Homebrew/brew/pull/20192/files#diff-79ec0566b38c06e8f122779717fe43cd20964da8c1998018082a1ca50ad24257